guac_core/src/crypto.rs

```diff
@@ -1,34 +1,100 @@
+use clarity::abi::derive_signature;
+use clarity::Transaction;
 use clarity::{Address, PrivateKey, Signature};
+use failure::Error;
 use multihash::{encode, Hash};
+use network::Web3Handle;
 
-use owning_ref::RwLockWriteGuardRefMut;
+use owning_ref::{RwLockReadGuardRef, RwLockWriteGuardRefMut};
 use sha3::{Digest, Keccak256};
 use std::sync::{Arc, RwLock};
 
+use error::GuacError;
+use futures::future::ok;
+use futures::Future;
+use futures::IntoFuture;
+use futures::Stream;
 use num256::Uint256;
+use std::time;
+use web3::types::{Bytes, FilterBuilder, Log};
 
 /// A global object which stores per node crypto state
 lazy_static! {
     pub static ref CRYPTO: Arc<RwLock<Crypto>> = Arc::new(RwLock::new(Crypto::new()));
 }
 
+pub struct Config {
```

Comment:
Gauc is supposed to get it's config from rita, is that viable?

Reply:
Yes. Rita would have to call `CRYPTO.init` with properties received from config file.